File: mmo-site/src/models/order.ts
export interface OrderRequest {
    productId: number;
    couponId: number;
    quantity: number;
}

export interface OrderResponse {
    id: string;
    userId: number;
    product: {
        id: number;
        name: string;
        description: string;
    },
    orderCode: string;
    quantity: number;
    amount: number;
    accounts: string;
    createdAt: string;
    updatedAt: string;
    isDeleted: boolean;
}

File: mmo-api/src/main/java/com/example/mmoapi/entity/Order.java
package com.example.mmoapi.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "orders")
public class Order {
    @Id
    private String id;
    private Long userId;
    private Long productId;
    private Long couponId;

    @Indexed(unique = true)
    private String orderCode;

    private int quantity;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean isDeleted;
}
